OSDOCS-16081 created release notes for 0.2.0 (#98607)
openshift/openshift-docs: security/zero_trust_workload_identity_manager/zero-trust-manager-release-notes.adoc

:_mod-docs-content-type: ASSEMBLY
[id="zero-trust-manager-release-notes"]
= Zero Trust Workload Identity Manager release notes
d
include::_attributes/common-attributes.adoc[]
:context: zero-trust-manager-release-notes

toc::[]

The {zero-trust-full} leverages Secure Production Identity Framework for Everyone (SPIFFE) and the SPIFFE Runtime Environment (SPIRE) to provide a comprehensive identity management solution for distributed systems. {zero-trust-full} supports SPIRE version 1.12.4 running as an operand.

These release notes track the development of {zero-trust-full}.

[id="zero-trust-manager-release-notes-2"]
== {zero-trust-full} 0.2.0 (Technology Preview)

Issued: 2025-09-08

There are no advisories for this release.

[id="zero-trust-manager-features-enhancements_{context}"]
=== New features and enhancements

*Support for managed OIDC Routes*

With this release, the Operator exposes the SPIRE OIDC Discovery Provider endpoints through OpenShift Routes under the domain `*.apps.<cluster-domain>` for the selected default installation.

> With this release, the managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscovetProvider` spec.

Author: Done

> With this release, the managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscovetProvider` spec.

Author: Done

With this release, the `managedRoute` field is boolean and is set to `true` by default. If set to `false`, the Operator stops managing the route and the existing route is not automatically deleted. If set back to 'true', the Operater resumes managing the route. If a route does not exist, the Operator creates a new one. If a route already exists, the operator forces an update to resolve any conflicts.

> With this release, the `externalSecretRef` references an externally managed Secret that contians the TLS certificate for the `oidc-discovery-provider` Route host. When provided, this populates the route's `.Spec.TLS.ExternalCertificate` field.

Author: Done

* SpireServer

* ZeroTrustWorkloadIdentityManager

* SpireAgents

* SpireOIDCDiscoveryProvider

With this release, the annotation needs to be added on the {zero-trust-full} resource for static resources such as Role-Based Access Control (RBAC), Switched Virtual Circuit (SVC), and SA resources.

With this release, the annotation should be added to the SpireServer, Agent, OIDC, and the Container Storage Interface (CSI) resource APIs.

With this release, once the annotation is removed and the pod restarts, the operator tries to come back to the desired state. The annotation is applied only once during start up or a restart.

[id="cert-manager-operator-0-2-0-known-issues_{context}"]
=== Known issues

* The `JwtIssuer` field for both the `SpireServer` and the OpenID Connect (OIDC) Discovery Provider needs to be a URL. If the `JwtIssuer` field is not in this format, an error displays.

> *Enabling the custom Certificat Authority Time-To-Live for the SPIRE bundle*

Author: Done

With this release, the following Time-To-Live (TTL) fields have been added to the `SpireServerSpec` API for SPIRE Server certificate management:

* `CAValidity` (default: 24h)

* `DefaultX509Validity` (default: 1h)

*  `DefaultJWTValidity` (default: 5m)

> The hard-coded values can be replaced in the server configuration with user-configurable options that provide users with the flexibility to customize certificate and {svid-full} lifetimes based on their security requirements.

Author: Done

With this release, the `externalSecretRef` references an externally managed Secret that contians the TLS certificate for the `oidc-discovery-provider` Route host. When provided, this populates the route's `.Spec.TLS.ExternalCertificate` field.

*Enable Manual User Configurations*

With this release, the Operator controller switches to `create-only` mode once the `ztwim.openshift.io/create-only=true` annotation is present on the Operator’s APIs. This allows resource creation while skipping the updates. A user can update the resources manually to test their configuration. This annotation needs to be added to the following APIs:

* SpireServer

* ZeroTrustWorkloadIdentityManager

* SpireAgents

* SpireOIDCDiscoveryProvider

With this release, the annotation needs to be added on the {zero-trust-full} resource for static resources such as Role-Based Access Control (RBAC), Switched Virtual Circuit (SVC), and SA resources.

> With this release, the annotation should be added to the SpireServer, Agent, OIDC, and the Container Storage Interface (CSI) resource APIs.

Author: Done

toc::[]

The {zero-trust-full} leverages Secure Production Identity Framework for Everyone (SPIFFE) and the SPIFFE Runtime Environment (SPIRE) to provide a comprehensive identity management solution for distributed systems. {zero-trust-full} supports SPIRE version 1.12.4 running as an operand.

These release notes track the development of {zero-trust-full}.

> [id="zero-trust-manager-release-notes-2"]

Reviewer: should the `2` be `0-2-0`?
```suggestion
[id="zero-trust-manager-release-notes-0-2-0"]
```

== {zero-trust-full} 0.2.0 (Technology Preview)

Issued: 2025-09-08

> There following advisories are available for the {zero-trust-full}.

Author: Done

* https://access.redhat.com/errata/RHBA-2025:15426[RHBA-2025:15426]
* https://access.redhat.com/errata/RHBA-2025:15427[RHBA-2025:15427]
* https://access.redhat.com/errata/RHBA-2025:15425[RHBA-2025:15425]
* https://access.redhat.com/errata/RHBA-2025:15428[RHBA-2025:15428]

This release of {zero-trust-full} is a Technology Preview.

> [id="zero-trust-manager-features-enhancements_{context}"]

Reviewer: This ID should probably include the version so that any further New features and enhancements sections have unique IDs, something like:
```suggestion
[id="zero-trust-manager-0-2-0-features-enhancements_{context}"]
```

=== New features and enhancements

> *Support for the managed OIDC Discovery Provider Route*

Reviewer: We're in sort of a weird transitional state wrt RN modules and headings premigration, but in general we should not be using bold as a heading level ([or anything other than a UI element, really](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference)). Change the bold headings to the next level down (`====`) and give them IDs. When we come back to modularize rel notes properly, these heading levels will feed the new structure.

```suggestion
[id="zero-trust-manager-0-2-0-features-oidc-discovery_{context}"]
==== Support for the managed OIDC Discovery Provider Route
```

Author: Done

With this release:

* the Operator exposes the `SPIREOIDCDiscoveryProvider` spec through OpenShift Routes under the domain `*.apps.<cluster_domain>` for the selected default installation.

* the `managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscoveryProvider` spec.

* the `managedRoute` field is boolean and is set to `true` by default. If set to `false`, the Operator stops managing the route and the existing route will not be deleted automatically. If set back to 'true', the Operator resumes managing the route. If a route does not exist, the Operator creates a new one. If a route already exists, the operator will override the user configuration if a conflict exists.

> * the `externalSecretRef` references an externally managed Secret that has the TLS certificate for the `oidc-discovery-provider` Route host. When provided, this populates the route's `.Spec.TLS.ExternalCertificate` field. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html-single/ingress_and_load_balancing/index#nw-ingress-route-secret-load-external-cert_secured-routes[Creating a route with externally managed certificate]

Author: So I redid this section. Let me know if the that's ok

toc::[]

The {zero-trust-full} leverages Secure Production Identity Framework for Everyone (SPIFFE) and the SPIFFE Runtime Environment (SPIRE) to provide a comprehensive identity management solution for distributed systems. {zero-trust-full} supports SPIRE version 1.12.4 running as an operand.

These release notes track the development of {zero-trust-full}.

[id="zero-trust-manager-release-notes-2"]
== {zero-trust-full} 0.2.0 (Technology Preview)

Issued: 2025-09-08

There following advisories are available for the {zero-trust-full}.

* https://access.redhat.com/errata/RHBA-2025:15426[RHBA-2025:15426]
* https://access.redhat.com/errata/RHBA-2025:15427[RHBA-2025:15427]
* https://access.redhat.com/errata/RHBA-2025:15425[RHBA-2025:15425]
* https://access.redhat.com/errata/RHBA-2025:15428[RHBA-2025:15428]

This release of {zero-trust-full} is a Technology Preview.

[id="zero-trust-manager-features-enhancements_{context}"]
=== New features and enhancements

> *Support for managed OIDC Routes*

Reviewer 2: ```suggestion
*Support for managed OIDC Discovery Provider Route*
```

Author: Done

With this release:

* the Operator exposes the `SPIREOIDCDiscoveryProvider` spec through OpenShift Routes under the domain `*.apps.<cluster-domain>` for the selected default installation.

> * the managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscoveryProvider` spec.

Reviewer 2: ```suggestion
* the `managedRoute` and  `externalSecretRef` fields have been added to the `spireOidcDiscoveryProvider` spec.
```

> * the `managedRoute` field is boolean and is set to `true` by default. If set to `false`, the Operator stops managing the route and the existing route will not be deleted automatically. If set back to 'true', the Operater resumes managing the route. If a route does not exist, the Operator creates a new one. If a route already exists, the operator will override the user configuration if a conflict exists. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html-single/ingress_and_load_balancing/index#nw-ingress-route-secret-load-external-cert_secured-routes[Creating a route with externally managed certificate]
> 
> * the `externalSecretRef` references an externally managed Secret that contians the TLS certificate for the `oidc-discovery-provider` Route host. When provided, this populates the route's `.Spec.TLS.ExternalCertificate` field.

Reviewer 2: ```suggestion
* the `managedRoute` field is boolean and is set to `true` by default. If set to `false`, the Operator stops managing the route and the existing route will not be deleted automatically. If set back to 'true', the Operater resumes managing the route. If a route does not exist, the Operator creates a new one. If a route already exists, the operator will override the user configuration if a conflict exists.

* the `externalSecretRef` references an externally managed Secret that contians the TLS certificate for the `oidc-discovery-provider` Route host. When provided, this populates the route's `.Spec.TLS.ExternalCertificate` field. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html-single/ingress_and_load_balancing/index#nw-ingress-route-secret-load-external-cert_secured-routes[Creating a route with externally managed certificate]
```

*Enabling the custom Certificate Authority Time-To-Live for the SPIRE bundle*

With this release:

> * the following Time-To-Live (TTL) fields have been added to the `SpireServerSpec` API for SPIRE Server certificate management:

Reviewer 2: ```suggestion
* the following Time-To-Live (TTL) fields have been added to the `SpireServer` Custom Resource API for SPIRE Server certificate management:
```

** `CAValidity` (default: 24h)

** `DefaultX509Validity` (default: 1h)

> **  `DefaultJWTValidity` (default: 5m)

Reviewer 2: ```suggestion
** `DefaultJWTValidity` (default: 5m)
```

** SpireOIDCDiscoveryProvider

** ZeroTrustWorkloadIdentityManager

* the annotation needs to be added on the {zero-trust-full} resource for static resources such as Role-Based Access Control (RBAC), service resources, and `serviceAccount` resources.

* Once the annotation is removed and the pod restarts, the operator tries to come back to the desired state. The annotation is applied only once during start up or a restart.

*Enabling the custom Certificate Authority Time-To-Live for the SPIRE bundle*

With this release:

* the following Time-To-Live (TTL) fields have been added to the `SpireServerSpec` API for SPIRE Server certificate management:

** `CAValidity` (default: 24h)

** `DefaultX509Validity` (default: 1h)

> **  `DefaultJWTValidity` (default: 5m)

Reviewer 2: ```suggestion
** `DefaultJWTValidity` (default: 5m)
```

> * the annotation needs to be added on the {zero-trust-full} resource for static resources such as role-based access control (RBAC), service resources, and `serviceAccount` resources.

Reviewer 2: Agree.

* Once the annotation is removed and the pod restarts, the operator tries to come back to the required state. The annotation is applied only once during start or a restart.

[id="cert-manager-operator-0-2-0-known-issues_{context}"]

> === Known issues

Reviewer 3: This should not be called out as a known issue. it is how the updated implementation is working.

> * The `JwtIssuer` field for both the `SpireServer` and the OpenID Connect (OIDC) Discovery Provider needs to be a URL. If the `JwtIssuer` field is not in this format, an error displays.

Reviewer 3: ```suggestion
* The `JwtIssuer` field for both the `SpireServer` and the `SpireOIDCDiscoveryProvider` needs to be a URL. If the `JwtIssuer` field is not in this format, an error displays.
```
